plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/StandardProjectWizard.java

```diff
@@ -58,10 +57,26 @@ public boolean performFinish() {
       status = new Status(Status.ERROR, "todo plugin ID", errorCode, ex.getMessage(), null);
     }
     
-    // todo if fail, call setErrorMessage()
+    showErrorMessageIfNecessary(status);
+    
     return status.isOK();
   }
 
+  
+  // visible for testing
+  void showErrorMessageIfNecessary(IStatus status) {
+    // if fail, display error message in wizard
+    if (status == Status.CANCEL_STATUS) {
+      page.setErrorMessage("User canceled project creation");
+    } else if (!status.isOK()) {
+      String message = "Failed to create project";
+      if (status.getMessage() != null && !status.getMessage().isEmpty()) {
+        message += ": " + status.getMessage();
+      }
+      page.setErrorMessage(message);
```

Comment:
OK, I've changed this to use StatusManager.getManager().handle(status, StatusManager.SHOW);